### ISSUE TITLE
Type of property unknown from within the class itself

### ISSUE THREAD

**Issue by dbrattli** [microsoft/pyright#3488]
**Describe the bug**

In pyright 1.1.2.47 it looks like properties on a class is not visible to the members of the class itself.

![image](https://user-images.githubusercontent.com/849479/169479103-3f210d00-e262-480a-805f-a0ba92124446.png)

**To Reproduce**
Steps to reproduce the behavior. See minimal repro below

**Expected behavior**

I would expect the property to have value with a valid type.

**Screenshots or Code**
If applicable, add screenshots or the text of the code (surrounded by triple back ticks) to help explain your problem.
```python
from typing import Generic, Iterable, TypeVar

_T = TypeVar("_T")


class Test(Generic[_T]):
    def __init__(self, value: _T) -> None:
        self._value = value

    @property
    def value(self) -> _T:
        return self._value

    def values(self) -> Iterable[_T]:
        return [self.value]
```

**VS Code extension or command-line**

Pylance: v2022.5.2, Pyright 1.1.247

**Additional context**

First seen in 1.1.247

**Comment by erictraut** [comment]
Thanks for the bug report. This will be addressed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.248, which I just published. It will also be included in the next release of pylance.
